test/unit/org/apache/cassandra/index/sai/cql/IndexCompressionTest.java

```diff
@@ -75,25 +77,14 @@ public void testKeyCompression()
     }
 
     @Test
-    public void testKeyCompressionIsCommonForAllIndexes()
+    public void testKeyCompressionMustMatchOnAllIndexes()
     {
         // When creating another index with different key compression, the key compression of the first index
         // gets overwritten by the compression settings of the second index.
         // This is beacuse both indexes share the same primary key map, and it can be compressed in one way only.
         createTable("CREATE TABLE %s (pk int, c text, val1 text, val2 text, PRIMARY KEY(pk, c))");
-        ClientWarn.instance.captureWarnings();
-        String index1 = createIndex("CREATE CUSTOM INDEX ON %s(val1) USING 'StorageAttachedIndex' WITH key_compression = {'class': 'LZ4Compressor'}");
-        Assertions.assertThat(ClientWarn.instance.getWarnings()).isNull();
-
-        String index2 = createIndex("CREATE CUSTOM INDEX ON %s(val2) USING 'StorageAttachedIndex' WITH key_compression = {'class': 'ZstdCompressor'}");
-        Assertions.assertThat(ClientWarn.instance.getWarnings()).hasSize(1).allMatch(e -> e.contains(index1));
-
-        ColumnFamilyStore cfs = getCurrentColumnFamilyStore();
-        IndexMetadata index1Metadata = cfs.indexManager.getIndexByName(index1).getIndexMetadata();
-        IndexMetadata index2Metadata = cfs.indexManager.getIndexByName(index2).getIndexMetadata();
-
-        assertTrue(index2Metadata.keyCompression.getSstableCompressor() instanceof ZstdCompressor);
-        assertTrue(index1Metadata.keyCompression.getSstableCompressor() instanceof ZstdCompressor);
+        createIndex("CREATE CUSTOM INDEX ON %s(val1) USING 'StorageAttachedIndex' WITH key_compression = {'class': 'LZ4Compressor'}");
+        assertThrows(InvalidRequestException.class, () -> createIndex("CREATE CUSTOM INDEX ON %s(val2) USING 'StorageAttachedIndex' WITH key_compression = {'class': 'ZstdCompressor'}"));
```

Comment:
Nit: We could check the exception message since `InvalidRequestException` is kind of a catch-all exception.

Reply:
yup